### ISSUE TITLE
[Metrics API] Add support for histogram advice API

### ISSUE THREAD

**Issue by lalitb** [open-telemetry/opentelemetry-cpp#2132]
Experimental histogram advice API to enable instrumentation library suggest histogram buckets - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice

**Comment by marcalff** [comment]
Adding the breaking change label: this will require a new or changed API, hence changing the API ABI for metrics.

**Comment by dufferzafar** [comment]
Is the Advice / Hints API still experimental? https://github.com/open-telemetry/opentelemetry-specification/issues/2229

The go client already has support for it now: https://github.com/open-telemetry/opentelemetry-go/pull/4605

**Comment by Falmarri** [comment]
The java client as well. Not really sure how to use histograms at all without that

**Comment by lalitb** [comment]
@Falmarri - The histogram buckets can be still configured using View API. Or let us know if you are looking for specific configuration from advisory API.

**Comment by Falmarri** [comment]
@lalitb Did I miss docs on this? Is there an example? This is using the `sdk` classes, right?

**Comment by lalitb** [comment]
Yes, there are unit-tests for custom buckets - https://github.com/open-telemetry/opentelemetry-cpp/blob/fabd8cc2bc318cb47d5db7322ea9c8cd3f4b847a/sdk/test/metrics/histogram_test.cc#L78

**Comment by BugelNiels** [comment]
I agree that it would be nice to have support for `explicitBucketBoundaries`: https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-advisory-parameter-explicitbucketboundaries
The definitions of the instruments and the setup for exporting metrics typically don't live in the same place, which means that in these cases it does not seem possible to have the histogram instrument definition and its boundaries defined in a single place. By adding `explicitBucketBoundaries` to the histogram creation API, this would be possible
